Objective-C/Internal/Replicator/CBLWebSocket.mm

```diff
@@ -443,13 +462,37 @@ - (void) receivedProxyHTTPResponse: (CFHTTPMessageRef)httpResponse {
 - (void) receivedHTTPResponse: (CFHTTPMessageRef)httpResponse {
     // Post the response headers to LiteCore:
     NSDictionary *headers =  CFBridgingRelease(CFHTTPMessageCopyAllHeaderFields(httpResponse));
+    
+    // save to lite core
     NSString* cookie = headers[@"Set-Cookie"];
-    if ([cookie containsString: @", "]) {
-        // CFHTTPMessage incorrectly merges multiple Set-Cookie headers. Undo that:
-        NSMutableDictionary* newHeaders = [headers mutableCopy];
-        newHeaders[@"Set-Cookie"] = [cookie componentsSeparatedByString: @", "];
-        headers = newHeaders;
+    if (cookie.length > 0) {
+        C4Error err = {};
+        if (!c4db_setCookie(_c4db, cookie.c4slice, _remoteURL.host.c4slice,
```

Comment:
It seems like c4db_setCookie handles only one cookie and CFHTTPMessage merged all into one. We need to break them first before set the cookie.